app/controllers/HomeController.scala

```diff
@@ -31,12 +34,44 @@ import scala.concurrent.Future
 class HomeController @Inject()(val checkSessionTimeout: SessionTimeoutPredicate,
                                val authenticate: AuthenticationPredicate,
                                val retrieveNino: NinoPredicate,
+                               val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                               val reportDeadlinesService: ReportDeadlinesService,
+                               val calculationService: CalculationService,
+                               val itvcErrorHandler: ItvcErrorHandler,
+                               val financialTransactionsService: FinancialTransactionsService,
                                val itvcHeaderCarrierForPartialsConverter: ItvcHeaderCarrierForPartialsConverter,
                                implicit val config: FrontendAppConfig,
                                val messagesApi: MessagesApi) extends FrontendController with I18nSupport {
 
-  val home: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino).async {
-    implicit user => Future.successful(Ok(views.html.home()))
+
+  val home: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino andThen retrieveIncomeSources).async {
+    implicit user =>
+
+      reportDeadlinesService.getNextDeadlineDueDate(user.incomeSources).flatMap { latestDeadlineDate =>
+
+        calculationService.getAllLatestCalculations(user.nino, user.incomeSources.orderedTaxYears) flatMap {
+          case lastTaxCalcs if lastTaxCalcs.nonEmpty =>
+            Future.sequence(lastTaxCalcs.filter(_.isCrystallised).map { crystallisedTaxCalc =>
+              financialTransactionsService.getFinancialTransactions(user.mtditid, crystallisedTaxCalc.year) map { transactions =>
+                (crystallisedTaxCalc, transactions)
+              }
+            }).map { billDetails =>
+              billDetails.filter(_._2 match {
```

Comment:
If there are any 500+ responses we should be throwing an error and not loading the page